render/image.go

```diff
@@ -380,3 +421,72 @@ func boardYToDrawY(dc *gg.Context, y int) float64 {
 	// of (0,0) being the top left.
 	return float64((dc.Height() - BoardBorder*2 - SquareSizePixels) - (y * SquareSizePixels)) // flip!
 }
+
+type svgManager struct {
+	baseDir string
+}
+
+func (sm svgManager) ensureSubdirExists(subDir string) error {
+	path := filepath.Join(sm.baseDir, subDir)
+	_, err := os.Stat(path)
+	if errors.Is(err, fs.ErrNotExist) {
+		return os.MkdirAll(path, os.ModePerm)
+	}
+
+	return err
+}
+
+func (sm svgManager) loadSVGImageAsset(name string, aType AssetType, w, h int, rot rotations) (image.Image, error) {
```

Comment:
Cool cool cool. This is great, let me see if I'm following...

* SVG Manager pulls SVG sources from media and stores on disk...? for the lifetime of the container?
* It then rasterizes PNGs for width, height, rotation and caches in-mem (short term)
* loadSVGImageAsset actually returns a fully rasterized Image object (I think that tripped me up, but I get it)
* Colorization still happens at composition time, so source .PNGs are only cached unique on source/size/orientation, but not color

Am I following that correctly?

FWIW, I'm wondering if some of these abstractions should actually live inside the media layer, ie: when requesting a source image from media, I can request it in PNG or SVG format (probably need to add size and orientation to that too). Then disk mgmt, caching, etc. remains all inside media. Thoughts?

Reply:
Yep, you followed it right.

Also yeah, I think `media` would make sense as you explained :+1: